File: RetailCore.Domain/Entities/Product.cs
using System;

namespace RetailCore.Domain.Entities;

public class Product: BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public int Stock { get; set; } = 0;
    public Guid CategoryId { get; set; }
    public string? ThumbnailUrl { get; set; }
    public bool IsFeatured { get; set; } = false;
    public bool IsActive { get; set; } = true;
}

File: retail-admin-dashboard/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  slug: string;
  description?: string;
  price: number;
  stock: number;
  categoryName?: string; 
  categoryId?: string;
  thumbnailUrl?: string;
  images?: { id: string; url: string; altText?: string }[];
  isFeatured?: boolean;
  isActive?: boolean;
}

export interface ProductCreateRequest {
  name: string;
  description?: string;
  slug: string;
  price: number;
  stock: number;
  categoryId: string;
  thumbnailUrl?: string;
  isFeatured: boolean;
  images?: File[]; 
  attributes?: string;
}

export interface ProductUpdateRequest extends ProductCreateRequest {
  id: string;
  isActive: boolean;
}